include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp

```diff
@@ -159,10 +127,17 @@ class aligned_sequence_builder
         static_assert(std::same_as<value_type_t<trace_path_t>, trace_directions>,
                       "The value type of the trace path must be seqan3::detail::trace_directions");
 
+        // helper function to transform the coordinate into a pair
+        auto to_pair = [] (matrix_coordinate const coordinate) constexpr
+        {
+            return std::pair{coordinate.col, coordinate.row};
+        };
+
+        result_type res{};
         auto trace_it = std::ranges::begin(trace_path);
-        matrix_coordinate path_end = trace_it.coordinate();
+        std::tie(res.cut_interval_first.second, res.cut_interval_second.second) = to_pair(trace_it.coordinate());
```

Comment:
Instead I would propose offering a tuple_conversion inside of the matrix coordinate.
First, I don't like the tuple interface  because`get<0/1>` does not tell me anything. Second, I don't want to mix things in one class. So either it is a tuple or it is something with it's own functionality.